synapse/util/logcontext.py

```diff
@@ -93,6 +118,39 @@ def copy_to(self, record):
         for key, value in self.__dict__.items():
             setattr(record, key, value)
 
+        record.ru_utime, record.ru_stime = self.get_resource_usage()
+
+    def start(self):
+        if threading.current_thread() is not self.main_thread:
```

Comment:
https://docs.python.org/2/library/threading.html#threading.Thread
The wording makes it fairly clear that there is "a main thread object". So I'm fairly happy to use a "is same object" comparison here.